src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -1975,7 +1974,6 @@ static PyObject* section_getattro(NPySecObj* self, PyObject* pyname) {
     Section* sec = self->sec_;
     CHECK_SEC_INVALID(sec);
     PyObject* rv;
-    auto _pyname_tracker = nb::borrow(pyname);  // keep refcount+1 during use
     Py2NRNString name(pyname);
```

Comment:
I had a quick look at `Py2NRNString`; from what I can tell won't consume `pyname`, but can you confirm?

Reply:
After looking through `Py2NRNString` I'm inclined to not touch the reference counting of `pyname` right now. The reasons being:

* I'm unable to see any reason why the `nb::borrow` is wrong (slightly inefficient: yes).
* I can't quickly and confidently analyze `Py2NRNString` to confirm with certainty that it's not relying on the reference count being `+1`.
* I think all of `Py2NRNString` needs to be modernized.
* It's a small number of lines of code changed. 

Therefore, I feel it's not worth risking a critical bug will require out immediate attention. The reward is just too low in my opinion. I've asked if @alkino would try to modernize `Py2NRNString` for us. Afterwards it'll likely be trivial to understand that the borrow is pointless and safe to remove.

Reply:
I guess with "Consume" you don't mean taking ownership of the internal buffer, because incref would also not help in that case. By what I see in `Py2NRNString` it strdup, which is alright, BUT it *may* decref which is kinda scary...